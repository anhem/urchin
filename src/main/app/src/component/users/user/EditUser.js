import React, {Component} from "react";
import history from "../../../history";

class EditUser extends Component {

    update = (e) => {
        let data = {
            field: e.target.name,
            value: e.target.value
        };

        this.props.callbacks.setUser(data);
    };

    addGroup = () => {
        this.props.callbacks.addGroup(this.props.user.userId, this.props.user.groupId)
    };

    del = () => {
        this.props.callbacks.deleteUser(this.props.user.userId);
    };

    back = () => {
        history.push('/users');
    };

    render() {
        let username = this.props.user.username;
        let groupsForUser = this.props.groupsForUser.map((item, index) =>
            <li key={index.toString()}>
                <a href={`/groups/${item.groupId}`}>{item.groupName}</a>
            </li>
        );
        let availableGroups = this.props.availableGroups;

        return (
            <div>
                <h2>User</h2>
                <div>Username: {username}</div>
                <select
                    name="groupId"
                    onChange={this.update}>
                    <option> -- select group --</option>
                    {availableGroups.map(group =>
                        <option key={group.groupId} value={group.groupId}>{group.groupName}</option>
                    )}
                </select>
                <button onClick={this.addGroup}>Add Group</button>
                <button onClick={this.del}>Delete</button>
                <button onClick={this.back}>Back</button>
                <h2>Member of groups</h2>
                <ul>
                    {groupsForUser}
                </ul>
            </div>
        )
    }
}

export default EditUser