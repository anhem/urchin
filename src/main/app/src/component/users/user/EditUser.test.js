import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import EditUser from "./EditUser";

describe('EditUser', () => {

    let props = {
        user: {},
        groupsForUser: [],
        availableGroups: []
    };

    it('match snapshot', () => {
        expect(toJson(shallow(<EditUser {...props}/>))).toMatchSnapshot();
    });
});
