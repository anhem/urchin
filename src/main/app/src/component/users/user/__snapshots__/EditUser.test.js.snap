// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUser match snapshot 1`] = `
<div
  className="edit-user"
  data-view="editUser"
>
  <h2>
    User
  </h2>
  <div>
    Username: 
  </div>
  <select
    className="edit-user__available-groups"
    data-view="availableGroups"
    name="groupId"
    onChange={[Function]}
  >
    <option>
      -- select group --
    </option>
  </select>
  <button
    className="edit-user__add-group-btn"
    data-view="addGroup"
    onClick={[Function]}
  >
    Add Group
  </button>
  <button
    className="edit-user__delete-btn"
    data-view="delete"
    onClick={[Function]}
  >
    Delete
  </button>
  <button
    className="edit-user__back-btn"
    data-view="back"
    onClick={[Function]}
  >
    Back
  </button>
  <h2>
    Member of groups
  </h2>
  <ul
    className="edit-user__groups"
  >
    <li
      className="edit-user__groups__item"
      data-view="groups"
      key="0"
    >
      <Link
        to="/groups/11"
      >
        alreadyAddedGroup
      </Link>
      <button
        className="edit-user__groups__item__remove-group-btn"
        data-view="removeGroup"
        onClick={[Function]}
      >
        Remove Group
      </button>
    </li>
  </ul>
</div>
`;
